src/components/dialog/dialog.js

```diff
@@ -1,107 +1,293 @@
-        };
 
-        // Check if the dialog exists, return if not
-        if (DOM.dialog === undefined) {
-            return false;
+    /**
+     * @function handleBackdropClick
+     * @desc If backdrop has been clicked, dismiss dialog
+     * @param {Event} e
+     */
+    function handleBackdropClick(e) {
+        if (e.target === DOM.backdrop) {
+            hideDialog(state.currentDialog);
         }
+    }
+
+
+    /**
+     * @function createBackdrop
+     * @desc Creates the dialog backdrop
+     */
+    function createBackdrop() {
+        // Create the backdrop
+        DOM.backdrop = createEl({id: 'hawk', className: defaultClassNames.DIALOG_BACKDROP});
+
+        addBackdropA11y(DOM.backdrop);
+    }
 
-        // Remove backdrop if turned off
-        if (!showBackdrop) {
-            DOM.dialog.classList.add(defaultClassNames.NO_BACKDROP);
+
+    /**
+     * @function addBackdropA11y
+     * @desc Applies relevant roles and attributes to the backdrop
+     * @param {node} backdrop
+     */
+    function addBackdropA11y(backdrop) {
+        backdrop.setAttribute('aria-hidden', true);
+    }
+
+
+    /**
+     * @function addDialogA11y
+     * @desc Applies relevant roles and attributes to the dialog
+     * @param {node} dialog
+     */
+    function addDialogA11y(dialog) {
+        const role = isAlert ? 'alertdialog' : 'dialog';
+
+        dialog.setAttribute('aria-hidden', 'true');
+        dialog.setAttribute('role', role);
+    }
+
+
+    /**
+     * @function bindOpenEvents
+     * @desc Finds all open buttons and attaches click event listener
+     * @param {node} dialog
+     */
+    function bindOpenEvents(dialog) {
+        const id = dialog.getAttribute('id');
+
+        // Grab all buttons which open this instance of the modal
+        const openButtons = qa(`${openBtn}[aria-controls="${id}"]`);
+
+        openButtons.forEach(button => button.addEventListener('click', openDialog));
+    }
+
+
+    /**
+     * @function openDialog
+     * @desc sets up dialog and state ready to be shown
+     * @param {node} dialog
+     */
+    function openDialog(e) {
+        // Get trigger button so focus can be returned to it later
+        const button = e.target;
+        // Get dialog that should be opened
+        const dialog = document.getElementById(button.getAttribute('aria-controls'));
+
+        //  Update State
+        state.currentOpenButton = button;
+        state.currentDialog = dialog;
+
+        showDialog(dialog);
+    }
+
+
+    /**
+     * @function showDialog
+     * @desc Sets up focusable elements, close and key events and displays modal
+     */
+    function showDialog(dialog) {
+        //  Focus the modal and remove aria attributes
+        dialog.setAttribute('tabindex', 1);
+        dialog.setAttribute('aria-hidden', false);
+
+        //  Set the first and last focusable elements
+        state.focusableElements = qa(NATIVELY_FOCUSABLE_ELEMENTS.join(), dialog);
```

Comment:
Potentially, I think we should discuss what we do with page wide DOM nodes and instance (state) specific ones .  For now I've just updated the comment above `state` and `DOM` objects to explain better what they should contain